CRM/Upgrade/Form.php

```diff
@@ -840,6 +841,8 @@ public static function doCoreFinish(): bool {
    * @return bool
    */
   public static function doFinish(): bool {
+    $session = CRM_Core_Session::singleton();
+    $session->reset(2);
```

Comment:
Yes this makes sense and is better. I didn't actually look at what config->cleanupCaches does just copied from rebuildMenuAndCaches.